### PR TITLE
Improve warnings

### PR DESCRIPTION
**Pull request by chunfantasy**
This PR is due to improving the way it displays the warnings.
It provides clear tips for the user to handle exceptions at right time.
Warnings should be removed when they are not needed.

### CODE REVIEW

File: lib/actions/usbsdfuTargetActions.js [hunk 1]
```diff
@@ -415,6 +417,21 @@ function operateDFU(inputDfuImages, inputDevice) {
...
+                'The array of supported SoftDevices for the update does not ' +
+                'contain the FWID of the current SoftDevice.')) {
+                dispatch(warningActions.addUserWarning(
+                    'SdReq is not supported. Please select a correct one.'));
+            }
+
+            // Notify the user to choose writable Bootloader
+            if (error.message.includes('This bootloader requires DFU updates to be signed.')) {
+                dispatch(warningActions.addUserWarning(
+                    'This bootloader need to be signed. Your action is rejected.'));
```
Comment (bihanssen) [inline lib/actions/usbsdfuTargetActions.js]: Suggest rewording to something like "The bootloader update must be signed. The DFU operation has been cancelled."

File: lib/actions/usbsdfuTargetActions.js [hunk 2]
```diff
@@ -438,11 +455,15 @@ function operateDFU(inputDfuImages, inputDevice) {
...
                 if (error.message.startsWith('Timeout while waiting')) {
-                    logger.error('It seems that the APP does not have a Nordic Trigger Interface.');
-                    logger.error('Please physically reset the device by clicking the reset button if you want to verify the flash.');
-                    dispatch(warningActions.addWarning('Please physically reset device.'));
+                    logger.error('It seems that the APP does not have ' +
+                        'a Nordic DFU Trigger Interface.');
+                    logger.error('Please physically reset the device by ' +
+                        'clicking the reset button if you want to verify the flash.');
+                    dispatch(warningActions.addUserWarning(
+                        'Seems that Nordic DFU Trigger Interface not found. ' +
```
Comment (bihanssen) [inline lib/actions/usbsdfuTargetActions.js]: Suggest rewording to something like "It seems that the Nordic DFU Trugger Interface was not found. Please physically reset the device."

File: lib/actions/usbsdfuTargetActions.js [hunk 3]
```diff
@@ -485,9 +507,11 @@ export function write() {
         dispatch(operateDFU(dfuImages))
             .then(async device => {
                 // In case that DFU fails and no device is returned,
-                // a warning should be displayed.
+                // an error message should be displayed in log.
+                // Moreover, the error should be handled during operating DFU
+                // by giving a warning and introduction to the user.
                 if (!device) {
-                    dispatch(warningActions.addWarning('Device not found due to failue during DFU'));
+                    logger.error('Device not found due to failue during DFU');
```
Comment (bihanssen) [inline lib/actions/usbsdfuTargetActions.js]: /failue/failure/

File: lib/actions/usbsdfuTargetActions.js [hunk 4]
```diff
@@ -485,9 +507,11 @@ export function write() {
         dispatch(operateDFU(dfuImages))
             .then(async device => {
                 // In case that DFU fails and no device is returned,
-                // a warning should be displayed.
+                // an error message should be displayed in log.
+                // Moreover, the error should be handled during operating DFU
+                // by giving a warning and introduction to the user.
```
Comment (bihanssen) [inline lib/actions/usbsdfuTargetActions.js]: Replace "introduction" with "description"?

File: lib/actions/usbsdfuTargetActions.js (hunk 1 above)
Comment (bencefr) [inline lib/actions/usbsdfuTargetActions.js]: I suggest "Unsigned bootloader is not supported.", it is less offending.

File: lib/actions/usbsdfuTargetActions.js (hunk 2 above)
Comment (chunfantasy) [reply lib/actions/usbsdfuTargetActions.js]: Sentence has been simplified as "Nordic DFU Trugger Interface was not found. Please physically reset the device."
